stdlibs/stdlibs.go

```diff
@@ -20,6 +21,31 @@ func InjectNativeMappings(store gno.Store) {
 
 func InjectPackage(store gno.Store, pn *gno.PackageNode) {
 	switch pn.PkgPath {
+	case "internal/crypto/sha1":
+		pn.DefineNative("Sum",
+			gno.Flds( // params
+				"data", "[]byte",
+			),
+			gno.Flds( // results
+				"bz", "[]byte",
+			),
+			func(m *gno.Machine) {
+				arg0 := m.LastBlock().GetParams1().TV
+				bz := []byte(nil)
+				if arg0.V != nil {
+					slice := arg0.V.(*gno.SliceValue)
+					array := slice.GetBase(m.Store)
+					bz = array.GetReadonlyBytes()
```

Comment:
Please consider using 

```go
bz = array.GetReadonlyBytes()[:slice.Length]
```
The function `GetReadonlyBytes()` could return leading `0` in the bytes array.